File: src/data/services/local_storage_service.js
import { dummySessions } from '../../core/utils/dummy_data.js';

// In-memory session store. Replace with Firebase/Supabase/SQLite when ready.
const _sessions = [...dummySessions];

export function getSessions() {
  return [..._sessions];
}

export function addSession(session) {
  _sessions.unshift(session);
}

export function getSessionById(id) {
  return _sessions.find(s => s.id === id) || null;
}

export function clearSessions() {
  _sessions.length = 0;
}

File: lib/data/services/local_storage_service.dart
import '../models/gait_session.dart';
import '../../core/utils/dummy_data.dart';

class LocalStorageService {
  // In-memory storage — replace with shared_preferences or Hive for persistence
  final List<GaitSession> _sessions = List.from(dummySessions);

  Future<void> saveSession(GaitSession session) async {
    _sessions.insert(0, session);
  }

  Future<List<GaitSession>> loadHistory() async {
    return List.unmodifiable(_sessions);
  }

  Future<void> clearHistory() async {
    _sessions.clear();
  }
}
